ng/manage.ts

```diff
@@ -436,6 +436,72 @@ angular.module("armExplorer", ["ngRoute", "ngAnimate", "ngSanitize", "ui.bootstr
         window.location.href = "api/tenants/" + $scope.selectedTenant.id;
     };
 
+    $scope.resourceSearch = () => {
+        // trigger a refresh if needed, but do not need to wait for it to finish since cache should already exist
+        refreshResourceSearchCache();
+
+        var results: IResourceSearchSuggestion[] = [];
+        var keyword = $scope.resourceSearchModel.searchKeyword || "";
+        var stringCompare = (strA: string, strB: string): number=> {
+            if (strA > strB) {
+                return 1;
+            } else if (strA < strB) {
+                return -1;
+            } else {
+                return 0;
+            }
+        };
+
+        $scope.resourceSearchCache.data.forEach((item, index) => {
+            if (item.name.toLowerCase().indexOf(keyword.toLowerCase()) > -1
+                || item.type.toLowerCase().indexOf(keyword.toLowerCase()) > -1) {
+                results.push(<IResourceSearchSuggestion> item);
+            }
+        });
+
+        results.sort((a, b) => {
+            var result = stringCompare(a.type, b.type);
+            if (result === 0) {
+                return stringCompare(a.name, b.name);
+            }
+
+            return result;
+        });
+
+        $scope.resourceSearchModel.suggestions = results;
+        if ($scope.resourceSearchModel.suggestions.length > 0) {
+            $scope.resourceSearchModel.isSuggestListDisplay = true;
+        } else {
+            $scope.resourceSearchModel.isSuggestListDisplay = false;
+        }
+    };
+
+    $scope.delayResourceSearch = () => {
```

Comment:
Just an FYI - I'm using [Rx extensions ](https://github.com/Reactive-Extensions/RxJS) for managing these things. It's pretty easy to use. Here is a [sample for their angular bindings](https://github.com/Reactive-Extensions/rx.angular.js/blob/master/examples/%24createObservableFunction/app.js) 

if you were to use it here the code for `delayResouceSearch` would look like this (and it should have the exact same effect)

```javascript
$scope.$createObservableFunction("delayResourceSearch")
        .flatMapLatest(() => {
            return $timeout(() => {}, 300); // or any promise that resolves after the time you want
        }).subscribe(() => {
            $scope.resourceSearch();
        });
```